### DISCUSSION TITLE
We're looking for maintainers

### DISCUSSION THREAD

**Discussion by ewjoachim** [procrastinate-org/procrastinate#748] [Announcements]
For a few years, I've spent limited time maintaining Procrastinate. I don't find it thrilling to maintain the project alone, so I tend to spend my free time elsewhere and just take care of upgrading what needs to be upgraded so everything stays secure.

If there are other people willing to share design discussions, I'll be glad to share the burden, and I'll be glad to take some time to show you the in-and-outs of the project.

If you're interested, join the [Discord](https://discord.gg/JWZeNq6P6Z)

**Comment by jerluc** [comment]
@ewjoachim are you still looking for some help from folks? If it helps I'd be open to reviewing some issues/PRs and contributing the occasional bugfix/feature as needed.

FWIW I only have a few months of experience using procrastinate in production, so I'm not yet _super_ familiar with its internals, but I could probably pick things up given some guidance!

**Reply by ewjoachim** [reply]
Ah, definitely! If you want, I'd love to show you around (assuming you don't already know more than me 😄). We can try and find a time if you want this to be sync. Feel free to reach out via Discord if you use that (@ewjoachim).

In the meantime, I can try and answer questions you may have on existing issues.

Let's get this going! Thank you a lot ! <3

**Reply by ewjoachim** [reply]
(So, there's a discord server now: https://discord.gg/JWZeNq6P6Z . Feel free to join :) )

**Comment by zschumacher** [comment]
I'm happy to help out, love the idea of this project

**Reply by ewjoachim** [reply]
Great ! I think I'll setup a discord server with the (few) people who expressed their interest, so we can keep some kind of communication channel to organize ourselves, and get to know each other.

https://discord.gg/JWZeNq6P6Z
If Discord is ok with you, feel free to join. If you'd rather stick to GitHub, let's discuss how to onboard you :)

**Comment by pantlavanya** [comment]
I too love the idea of this project, can help out.

**Reply by ewjoachim** [reply]
Great :) Since this was posted, the project gained a co-maintainer and some contributors. That said, if you'd like to help out, please chime in in (well you guessed it) the discord server :)

**Comment by auvipy** [comment]
I like the projects Idea. I want to learn from this project and see how we could utilize it celery in future :)

**Reply by medihack** [reply]
Hi @auvipy, it's great to see you here! Before I started using Procrastinate and became a co-maintainer of the project, I worked with Celery for quite a while. Please feel free to ask any questions.

**Reply by auvipy** [reply]
Is it possible to use it as a back end of celery?

**Reply by medihack** [reply]
Do you mean as a broker backend? What must it implement for that? Are there any specifications (e.g., an abstract class to inherit)?

**Reply by auvipy** [reply]
Yes, like we got sqlalchemy broker now

**Reply by ewjoachim** [reply]
> [!NOTE]
> Don't read this as a way to refuse anything, but 100% as a mark of my lack of understanding of what we're talking about.

Is there a precedent of Celery brokers that are not just message passing solutions, but task processing libs? Like: is there a dramatiq broker for Celery?

I'm a little confused regarding what you're suggesting? Or would the aim be to use the message queue bit but not the worker or the task definition? In that case, it would mean, mostly, that Celery would just reuse the SQL schema and procedures? But then you're saying that there's a sqlalchemy backend, so somehow, Celery already has a working set of SQL queries that allows using postgres as a broker? (and other DBs too)? Or would the aim be to have a more specialized postgres broker that would use listen/notify and locks? I don't think the notion of locks exist in Celery's message protocol (unless I'm mistaken?) and I don't know if we could retrofit procrastinate periodic tasks in celery beat without changes so deep it would completely break the model?

**Comment by vladignatyev** [comment]
I want to help out! 

At first, I am going to use `procrastinate` in a small commercial project. We have a microservice that wraps some API that offers postponed processing of requests and we need a persistency for jobs which request this API. Using `procrastinate` could be a good starting point before diving into its' internals.

Obviously, I considered alternatives for my specific task: first, `Celery` and `rq`, they suit but they both are overkill for the microservice; `huey` seems to lack `asyncio` support. `procrastinate` looks much better and stands out by offering a more flexible approach. 

Please let me know if you still need contributors! 

I would also appreciate someone to help me get on board and point out some ideas and concepts beyond opened issues.
